app/services/read-only/service.go

```diff
@@ -117,6 +119,63 @@ func (s Service) FindAssetTransfer(
 	}
 }
 
+func (s Service) FindNftTransfer(
+	transferID string, tokenID string, serialNum int64, sender string, receiver string,
+	save func(transactionID, scheduleID, status string) error) {
+	for {
+		response, err := s.mirrorNode.GetNftTransactions(tokenID, serialNum)
+		if err != nil {
+			s.logger.Errorf("[%s] - Failed to get nft transactions after timestamp. Error: [%s]", transferID, err)
+			continue
+		}
+
+		finished := false
+		for _, transaction := range response.Transactions {
+			if transaction.Type == CryptoTransfer &&
+				transaction.ReceiverAccountID == receiver &&
+				transaction.SenderAccountID == sender {
+
+				isFound := false
+				scheduledTx, err := s.mirrorNode.GetScheduledTransaction(transaction.TransactionID)
+				if err != nil {
+					s.logger.Errorf("[%s] - Failed to retrieve scheduled transaction [%s]. Error: [%s]", transferID, transaction.TransactionID, err)
+					continue
+				}
+				for _, tx := range scheduledTx.Transactions {
+					if tx.Result == hedera.StatusSuccess.String() {
+						scheduleID, err := s.mirrorNode.GetSchedule(tx.EntityId)
+						if err != nil {
+							s.logger.Errorf("[%s] - Failed to get scheduled entity [%s]. Error: [%s]", transferID, scheduleID, err)
+							break
+						}
+						if scheduleID.Memo == transferID {
+							isFound = true
```

Comment:
You can move the logic if it is found here to skip 1 check. But also it can get too indented so not sure if it is better in terms of readability.